rasa/utils/io.py

```diff
@@ -121,7 +134,28 @@ def read_yaml(content: Text) -> Union[List[Any], Dict[Text, Any]]:
             .decode("utf-16")
         )
 
-    return yaml_parser.load(content) or {}
+    return content
+
+
+def change_sections_in_yaml_file(data: Dict[Text, Any], filename: Text) -> None:
+    """Changes specific sections (given by keys) in a yaml file.
+
+     Args:
+        data: The keys are the sections to be changed, the values the contents of the
+              respective sections.
+        filename: Name of the file to be changed.
+    """
+    old_content = read_file(filename, DEFAULT_ENCODING)
+
+    yaml_parser = _get_yaml_parser(typ="rt", add_version=False)
+    old_content = _fix_ascii_encoding(old_content)
+
+    new_content = yaml_parser.load(old_content) or {}
```

Comment:
we should consider merging this with `read_yml_file`

Reply:
It's structured differently now, check it out! :)
When wanting to preserve the format as well (indentation, comment positions etc.), the implementation got very specific and it didn't seem useful anymore to have this a utils function. So now the config specifics are done in `autoconfiguration.dump_config`, which calls (a slightly modified version of) `utils.io.read_yaml`.